pkg/operator/ceph/pool/validate.go

```diff
@@ -163,3 +165,47 @@ func ValidatePoolSpec(context *clusterd.Context, clusterInfo *cephclient.Cluster
 
 	return nil
 }
+
+// validateDeviceClasses validates the available device classes with the CephBlockPool HybridStorageSpec
+func validateDeviceClasses(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo,
+	p *cephv1.CephBlockPool, clusterName string) (map[string]string, error) {
+	statusInfo := make(map[string]string)
+	deviceClasses, err := cephclient.GetDeviceClasses(context, clusterInfo)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to fetch device classes in the cluster %q", clusterName)
+	}
+
+	if len(deviceClasses) == 0 {
+		logger.Debug("no device classes found in the cluster %q", clusterName)
+		return nil, nil
+	}
+
+	invalidDeviceClasses := []string{}
+	primaryDeviceClass := p.Spec.Replicated.HybridStorage.PrimaryDeviceClass
+	secondaryDeviceClass := p.Spec.Replicated.HybridStorage.SecondaryDeviceClass
+
+	if !contains(deviceClasses, primaryDeviceClass) {
+		invalidDeviceClasses = append(invalidDeviceClasses, primaryDeviceClass)
+	}
+	if !contains(deviceClasses, secondaryDeviceClass) {
+		invalidDeviceClasses = append(invalidDeviceClasses, secondaryDeviceClass)
+	}
+
+	if len(invalidDeviceClasses) > 0 {
+		msg := fmt.Sprintf("CephBlockPool %q has invalid device classes: %v", p.Name, invalidDeviceClasses)
```

Comment:
I don't think this is sufficient, we must check with `ceph osd crush class ls-osd $TYPE`, and each `type` must return at least one element.

Reply:
Thanks for pointing that out. Using `ceph osd crush class ls-osd $TYPE` to check if device class is valid or not.